### PR TITLE
clients: Add example WinForms GUI to TCP client

### PR DESCRIPTION
**Pull request by LecraM2003**
Create a simple GUI as an example how to use the client library with TCP on WindowsForms

Currently it only supports: connecting, disconnecting, steering and changing speed
Works with an Alphabot that is compatible with the new protocol (#38)

GUI:
image

### CODE REVIEW

**Comment by LecraM2003** [comment]
#duration: 4

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/AlphabotClientLibrary.Example.WinForms.csproj [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    <TargetFramework>net5.0-windows</TargetFramework>
+    <UseWindowsForms>true</UseWindowsForms>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\AlphabotClientLibrary.Core.Tcp\AlphabotClientLibrary.Core.Tcp.csproj" />
+    <ProjectReference Include="..\AlphabotClientLibrary.Core\AlphabotClientLibrary.Core.csproj" />
+  </ItemGroup>
+
+</Project>
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/AlphabotClientLibrary.Example.WinForms.csproj:14]:
```suggestion
</Project>

```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/AlphabotClientLibrary.Example.WinForms.csproj.user [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="Current" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup>
+    <Compile Update="MainForm.cs">
+      <SubType>Form</SubType>
+    </Compile>
+  </ItemGroup>
+</Project>
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/AlphabotClientLibrary.Example.WinForms.csproj.user:8]:
```suggestion
</Project>

```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.resx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  <resheader name="version">
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+</root>
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.resx:60]:
```suggestion
</root>

```

**Review by Filiprogrammer — CHANGES_REQUESTED** [review]
Use camelCase for form element names and change names like "label6" to something more indicative.

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Forms;
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.cs:16]:
```suggestion
using System;
using System.Net;
using System.Threading;
using System.Windows.Forms;
﻿using AlphabotClientLibrary.Core;
using AlphabotClientLibrary.Core.Tcp;
using AlphabotClientLibrary.Shared.Contracts;
using AlphabotClientLibrary.Shared.Models;
using AlphabotClientLibrary.Shared.Requests;
```

**Review by Filiprogrammer — CHANGES_REQUESTED** [review]

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.cs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            numSpeed.Enabled = false;
+        }
+
+        private void tbar_steer_Scroll(object sender, EventArgs e)
+        {
+            _steer = (sbyte)tbarSteer.Value;
+            lblSteer.Text = "Lenkung: " + _steer;
+        }
+
+        private void num_speed_ValueChanged(object sender, EventArgs e)
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.cs:76]:
```suggestion
        private void numSpeed_ValueChanged(object sender, EventArgs e)
```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.cs [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            _connectionHandler.Disconnect();
+            cbxConnected.Checked = false;
+            btnDisconnect.Enabled = false;
+            btnConnect.Enabled = true;
+            tbarSteer.Enabled = false;
+            numSpeed.Enabled = false;
+        }
+
+        private void tbar_steer_Scroll(object sender, EventArgs e)
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.cs:70]:
```suggestion
        private void tbarSteer_Scroll(object sender, EventArgs e)
```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.cs [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+                numSpeed.Enabled = true;
+            }
+            else
+            {
+                cbxConnected.Checked = false;
+                btnConnect.Enabled = true;
+            }
+        }
+
+        private void btn_disconnect_Click(object sender, EventArgs e)
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.cs:60]:
```suggestion
        private void btnDisconnect_Click(object sender, EventArgs e)
```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.cs [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+            InitializeComponent();
+            Thread thread = new Thread(UpdateSteer)
+            {
+                IsBackground = true
+            };
+
+            thread.Start();
+        }
+
+        private void btn_connect_Click(object sender, EventArgs e)
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.cs:38]:
```suggestion
        private void btnConnect_Click(object sender, EventArgs e)
```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+            // 
+            // btnConnect
+            // 
+            this.btnConnect.Location = new System.Drawing.Point(108, 187);
+            this.btnConnect.Name = "btnConnect";
+            this.btnConnect.Size = new System.Drawing.Size(112, 34);
+            this.btnConnect.TabIndex = 0;
+            this.btnConnect.Text = "Connect";
+            this.btnConnect.UseVisualStyleBackColor = true;
+            this.btnConnect.Click += new System.EventHandler(this.btn_connect_Click);
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs:58]:
```suggestion
            this.btnConnect.Click += new System.EventHandler(this.btnConnect_Click);
```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+            // btnDisconnect
+            // 
+            this.btnDisconnect.Enabled = false;
+            this.btnDisconnect.Location = new System.Drawing.Point(108, 228);
+            this.btnDisconnect.Name = "btnDisconnect";
+            this.btnDisconnect.Size = new System.Drawing.Size(112, 34);
+            this.btnDisconnect.TabIndex = 6;
+            this.btnDisconnect.Text = "Disconnect";
+            this.btnDisconnect.UseVisualStyleBackColor = true;
+            this.btnDisconnect.Click += new System.EventHandler(this.btn_disconnect_Click);
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs:116]:
```suggestion
            this.btnDisconnect.Click += new System.EventHandler(this.btnDisconnect_Click);
```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+            this.tbarSteer.Enabled = false;
+            this.tbarSteer.LargeChange = 10;
+            this.tbarSteer.Location = new System.Drawing.Point(314, 64);
+            this.tbarSteer.Maximum = 125;
+            this.tbarSteer.Minimum = -125;
+            this.tbarSteer.Name = "tbarSteer";
+            this.tbarSteer.Size = new System.Drawing.Size(316, 69);
+            this.tbarSteer.SmallChange = 5;
+            this.tbarSteer.TabIndex = 7;
+            this.tbarSteer.Scroll += new System.EventHandler(this.tbar_steer_Scroll);
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs:129]:
```suggestion
            this.tbarSteer.Scroll += new System.EventHandler(this.tbarSteer_Scroll);
```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs [hunk 12]
```diff
@@ -0,0 +1,248 @@
...
+            0});
+            this.numSpeed.Minimum = new decimal(new int[] {
+            50,
+            0,
+            0,
+            -2147483648});
+            this.numSpeed.Name = "numSpeed";
+            this.numSpeed.Size = new System.Drawing.Size(197, 31);
+            this.numSpeed.TabIndex = 8;
+            this.numSpeed.ValueChanged += new System.EventHandler(this.num_speed_ValueChanged);
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs:153]:
```suggestion
            this.numSpeed.ValueChanged += new System.EventHandler(this.numSpeed_ValueChanged);
```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs [hunk 13]
```diff
@@ -0,0 +1,248 @@
...
+            this.lblSteer.AutoSize = true;
+            this.lblSteer.Location = new System.Drawing.Point(651, 85);
+            this.lblSteer.Name = "lblSteer";
+            this.lblSteer.Size = new System.Drawing.Size(83, 25);
+            this.lblSteer.TabIndex = 13;
+            this.lblSteer.Text = "Lenkung:";
+            // 
+            // MainForm
+            // 
+            this.AccessibleDescription = "tbx_ip";
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs:202]:
```suggestion
```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs [hunk 14]
```diff
@@ -0,0 +1,248 @@
...
+            this.btnConnect.Name = "btnConnect";
+            this.btnConnect.Size = new System.Drawing.Size(112, 34);
+            this.btnConnect.TabIndex = 0;
+            this.btnConnect.Text = "Connect";
+            this.btnConnect.UseVisualStyleBackColor = true;
+            this.btnConnect.Click += new System.EventHandler(this.btn_connect_Click);
+            // 
+            // tbxPort
+            // 
+            this.tbxPort.AccessibleDescription = "tbx_port";
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs:62]:
```suggestion
```

File: clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs [hunk 15]
```diff
@@ -0,0 +1,248 @@
...
+            this.tbxPort.Location = new System.Drawing.Point(108, 119);
+            this.tbxPort.Name = "tbxPort";
+            this.tbxPort.Size = new System.Drawing.Size(150, 31);
+            this.tbxPort.TabIndex = 1;
+            this.tbxPort.Text = "9000";
+            // 
+            // tbxIp
+            // 
+            this.tbxIp.AccessibleDescription = "tbx_ip";
```
Comment (Filiprogrammer) [inline clients/AlphabotClientLibrary/AlphabotClientLibrary.Example.WinForms/MainForm.Designer.cs:71]:
```suggestion
```

**Review by Filiprogrammer — CHANGES_REQUESTED** [review]

**Review by Filiprogrammer — DISMISSED** [review]

**Pull request merged**